src/main/java/io/excaliburfrc/robot/subsystems/Climber.java

```diff
@@ -39,32 +84,59 @@ public void close() {
     }
   }
 
-  public void activateLeader(MotorMode m) {
-    motor.set(m.dutyCycle);
+  public void activateMotors(MotorMode m) {
+    leftMotor.set(m.dutyCycle);
+    rightMotor.set(m.dutyCycle);
   }
 
   public void openAngler() {
-    anglerPiston.set(DoubleSolenoid.Value.kForward);
+    piston.set(DoubleSolenoid.Value.kForward);
   }
 
   public void closeAngler() {
-    anglerPiston.set(DoubleSolenoid.Value.kReverse);
+    piston.set(DoubleSolenoid.Value.kReverse);
+  }
+
+  private Command reachSideHeight(
+      SparkMaxPIDController controller, RelativeEncoder encoder, double height) {
+    return new TrapezoidProfileCommand(
+            new TrapezoidProfile(
+                new TrapezoidProfile.Constraints(
+                    feedforward.maxAchievableVelocity(12, kA),
+                    feedforward.maxAchievableAcceleration(12, kV)),
+                new TrapezoidProfile.State(height, 0),
+                new TrapezoidProfile.State(encoder.getPosition(), 0)),
+            velocitySetpoint ->
+                controller.setReference(
+                    height,
+                    ControlType.kPosition,
+                    0,
+                    feedforward.calculate(velocitySetpoint.velocity),
+                    SparkMaxPIDController.ArbFFUnits.kVoltage),
+            this)
+        .withInterrupt(
+            () -> {
+              double position = encoder.getPosition();
+              return abs(height - position) <= THRESHOLD;
+            })
```

Comment:
TrapezoidProfileCommand finishes when the profile is finished, so the interrupt condition is redundant here.

Reply:
I put it for safety, in case that the trapezoid command keeps going for some reason

Reply:
Because the TrapezoidProfile relies only on the feedforward, if something it's not 100% accurate the command would can stop too late or too early, so I wanna add a ConditionalCommand that runs it again if it stopped to early

Reply:
The PID is there so that the profile is followed until the end...

Reply:
But the feedforward would be 0 in the last setReference

Reply:
The weight of the arm doesn't matter, just the force that pushes against the tip of the arm

Reply:
> But the feedforward would be 0 in the last setReference

why? and why is that relevant?

Reply:
The trapezoid creates some kind of simulation and updates the states right? So when the trapezoid is finished, it will set the velocity to 0, and then the feedforward will be 0, even though the pid might still be going